crates/typst/src/eval/call.rs

```diff
@@ -129,6 +129,18 @@ impl Eval for ast::FuncCall<'_> {
             (callee.eval(vm)?, args.eval(vm)?.spanned(span))
         };
 
+        // If the callee is a callable symbol, call its handler.
+        let callee = match callee {
+            Value::Symbol(sym) => {
+                if let Some(handler) = sym.func() {
+                    Value::Func(handler)
+                } else {
+                    Value::Symbol(sym)
+                }
+            }
+            callee => callee,
+        };
+
```

Comment:
Seeing this file changed, I'd like to note that I think the `Eval<FuncCall>::eval` function is really ugly in general, and deserves an overhaul. I'd be interested in discussing this if @laurmaedje agrees. I think splitting out the massive `let (...) = if { /* huge mess of 80+ lines */ } else {/* one single line */}` into a separate function would be a nice start, but there are a few early returns in the huge mess that make it complicated in the context of the overall function. (This is also somewhat relevant for the new VM refactor, since it completely redoes the func call system)

Reply:
Agree. When I saw this function, it was a bit hard to understand the whole behavior of it.

And I would suggest sending another PR to refactor it (let every PR do its own job).

Reply:
By the way, this change actually originates from #4299 on which this PR depends.

Reply:
@wrzian I agree as well. Not my proudest work 😂